### PR TITLE
Add Fortran support for dict initialisation

### PR DESCRIPTION
**Pull request by EmilyBourne**
Add Fortran support for dict initialisation in a similar way to lists. Fixes https://github.com/pyccel/pyccel/issues/1944

### CODE REVIEW

**Comment by EmilyBourne** [comment]
/bot run docs pylint pyccel_lint spelling coverage

File: pyccel/codegen/printing/fcode.py [hunk 1]
```diff
@@ -560,6 +560,31 @@ def _build_gFTL_module(self, expr_type):
...
+            elif isinstance(expr_type, DictType):
+                include = Import(LiteralString('map/template.inc'), Module('_', (), ()))
+                key_type = expr_type.key_type
+                value_type = expr_type.value_type
+                macros = []
+                if isinstance(key_type, FixedSizeNumericType):
+                    tmpVar_x = Variable(key_type, 'x')
+                    tmpVar_y = Variable(key_type, 'y')
+                    if isinstance(key_type.primitive_type, PrimitiveComplexType):
+                        lt_def = PyccelAssociativeParenthesis(PyccelLt(NumpyAbs(tmpVar_x), NumpyAbs(tmpVar_y)))
```
Comment (jalalium) [inline pyccel/codegen/printing/fcode.py:572]: The modulus is not enough to define a total order on complex numbers
the following code should output 2 but it outputs 1:
```python
if __name__ == '__main__':
    s = {0+1j : 5, 1+0j : 3}
    print(len(s))
```
the real and imaginary parts should also be compared

**Review by jalalium — CHANGES_REQUESTED** [review]

File: pyccel/codegen/printing/fcode.py (hunk 1 above)
Comment (EmilyBourne) [reply pyccel/codegen/printing/fcode.py]: Fixed in the same way as for sets: [fa8166d](https://github.com/pyccel/pyccel/pull/2005/commits/fa8166d47e66d9b880898803cc0db56fcf12de0e)
Test added that would have failed with the old method: 67c526d4d

**Review by jalalium — APPROVED** [review]
Well done!

**Pull request merged**